modules/integration/tests-integration/lazy-loading-test/src/test/java/org/wso2/appserver/integration/lazy/loading/LazyLoadingBaseTest.java

```diff
@@ -0,0 +1,506 @@
+        } catch (JSONException jsonException) {
+            String customErrorMessage = "JSONException when retrieving the values from json object WebAppStatus" +
+                    requestUrl + "\n" + jsonException.getMessage();
+            log.error(customErrorMessage);
+            throw new LazyLoadingTestException(customErrorMessage, jsonException);
+        }
+
+        return webAppStatus;
+
+    }
+
+
+    /**
+     * Check the given Jaggery application is deployed correctly. This method is wait for 90 seconds
+     * for deployment of jaggery application and each 500 milliseconds  of wait it will check the
+     * deployment status.
+     *
+     * @param appName Name of the application.
+     * @return true if the application is get deployed before the maximum wait time of 90 seconds.
+     * @throws LazyLoadingTestException
+     */
+    protected boolean isJaggeryAppDeployed(String appName) throws LazyLoadingTestException {
+        int deploymentDelay = 90 * 1000;
+        log.info("waiting " + deploymentDelay + " millis for Service deployment " + appName);
+        WebAppAdminClient webAppAdminClient;
+        try {
+            webAppAdminClient = new WebAppAdminClient(backendURL, sessionCookie);
+        } catch (AxisFault axisFault) {
+            String customErrorMessage = "AxisFault Exception  when creating WebAppAdminClient object. backend URL:" +
+                    backendURL + " Session Cookie: " + sessionCookie + "\n" + axisFault.getMessage();
+            log.error(customErrorMessage);
+            throw new LazyLoadingTestException(customErrorMessage, axisFault);
+        }
+        List<String> webAppList;
+        List<String> faultyWebAppList;
+
+        boolean isWebAppDeployed = false;
+        Calendar startTime = Calendar.getInstance();
+        long time;
+        boolean doLoop = true;
+        while (((time = (Calendar.getInstance().getTimeInMillis() - startTime.getTimeInMillis())) < deploymentDelay)
+                && doLoop) {
+            try {
+                webAppList = webAppAdminClient.getWebApplist(appName);
+                faultyWebAppList = webAppAdminClient.getFaultyWebAppList(appName);
+            } catch (RemoteException remoteException) {
+                String customErrorMessage = "remoteException Exception when calling methods in WebAppAdminClient\n" +
+                        remoteException.getMessage();
+                log.error(customErrorMessage);
+                throw new LazyLoadingTestException(customErrorMessage, remoteException);
+            }
+
+            for (String faultWebAppName : faultyWebAppList) {
+                if (faultWebAppName.equalsIgnoreCase(appName)) {
+                    isWebAppDeployed = false;
+                    log.info(appName + "- Jaggery Application is faulty");
+                    doLoop = false;
+                }
+            }
+
+            for (String name : webAppList) {
```

Comment:
don't use general names like "name" - try to use specific names which is relevant to the context